util/data/msgparse.c

```diff
@@ -1093,16 +1177,19 @@ parse_edns_from_pkt(sldns_buffer* pkt, struct edns_data* edns,
 	edns->ext_rcode = sldns_buffer_read_u8(pkt); /* ttl used for bits */
 	edns->edns_version = sldns_buffer_read_u8(pkt);
 	edns->bits = sldns_buffer_read_u16(pkt);
-	edns->opt_list = NULL;
+	edns->opt_list_in = NULL;
+	edns->opt_list_out = NULL;
+	edns->opt_list_modules_out = NULL;
 	edns->padding_block_size = 0;
 
 	/* take the options */
 	rdata_len = sldns_buffer_read_u16(pkt);
 	if(sldns_buffer_remaining(pkt) < rdata_len)
 		return LDNS_RCODE_FORMERR;
 	rdata_ptr = sldns_buffer_current(pkt);
-	if(!parse_edns_options(rdata_ptr, rdata_len, edns, region))
-		return LDNS_RCODE_SERVFAIL;
+	rcode = parse_edns_options(rdata_ptr, rdata_len, edns, cfg, c, region);
+	if(rcode)
+		return rcode;
 
 	/* ignore rrsigs */
 
```

Comment:
@TCY16 Could you please remove the return below too? And the `rcode` declaration? Thanks!
```suggestion
	/* ignore rrsigs */
	return parse_edns_options(rdata_ptr, rdata_len, edns, cfg, c, region);
```